integration-tests/test-cases/scenario-tests/src/test/java/io/cellery/integration/scenario/tests/EmployeePortalTestCase.java

```diff
@@ -94,4 +140,234 @@ public void cleanup() {
+     * @throws IOException              if sendGet method fails
+     * @throws KeyManagementException   if sendGet method fails
+     */
+    private String getHrApiId() throws NoSuchAlgorithmException, IOException
+            , KeyManagementException {
+        String hrApiId = "";
+        // Get the list of APIs
+        String url = "https://wso2-apim/api/am/store/v0.14/apis";
+        //Add headers
+        Map<String, String> headers = new HashMap<>();
+        String apis = httpClient.sendGet(url, headers);
+        JsonObject responseJson = new JsonParser().parse(apis).getAsJsonObject();
+        JsonArray apiList = responseJson.get("list").getAsJsonArray();
+        for (int i = 0; i < apiList.size(); i++) {
+            JsonObject jsonItem = apiList.get(i).getAsJsonObject();
+            // Get the id of hr api
+            if (jsonItem.get("name").toString().equals("\"hr_inst_global_1_0_0_hr_api\"")) {
+                hrApiId = jsonItem.get("id").toString().replaceAll("^\"|\"$", "");
+            }
+        }
+        return hrApiId;
+    }
+
+    /**
+     * Get the oauth token required to access wso2-apim.
+     *
+     * @param clientIdColonClientSecret A concatenated string comprising client id and client secret delimited by a colon
+     * @return oauth token
+     * @throws NoSuchAlgorithmException if sendPost method fails
+     * @throws IOException              if sendPost method fails
+     * @throws KeyManagementException   if sendPost method fails
+     */
+    private String getOauthToken(String clientIdColonClientSecret) throws NoSuchAlgorithmException, IOException
+            , KeyManagementException {
+        String passwordGrantType;
+        String url = "https://wso2-apim-gateway/token";
+        //Add headers
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Authorization", "Basic " + Base64.getEncoder().encodeToString(
+                clientIdColonClientSecret.getBytes(StandardCharsets.UTF_8)));
+
+        String payload = "grant_type=password&username=alice&password=alice123&scope=apim:subscribe";
+        String response = httpClient.sendPost(url, payload, headers);
+        JsonObject responseJson = new JsonParser().parse(response).getAsJsonObject();
+        Assert.assertTrue(responseJson.isJsonObject());
+        // Get the token from the response
+        passwordGrantType = responseJson.get("access_token").toString().replaceAll("^\"|\"$", "");
+        return passwordGrantType;
+    }
+
+    /**
+     * Get the id of default application.
+     *
+     * @param oauthToken An oauth token to access wso2-apim
+     * @return default application id
+     * @throws NoSuchAlgorithmException if sendGet method fails
+     * @throws IOException              if sendGet method fails
+     * @throws KeyManagementException   if sendGet method fails
+     */
+    private String getDefaultApplicationId(String oauthToken) throws NoSuchAlgorithmException, IOException
```

Comment:
This is also a potential method which could be reused.

Reply:
Updated and moved to util package.